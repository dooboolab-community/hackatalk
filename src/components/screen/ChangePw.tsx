import { Alert, EmitterSubscription, Keyboard, KeyboardEvent, SafeAreaView } from 'react-native';
import { Button, EditText } from '@dooboo-ui/native';
import React, {
  ReactElement,
  useEffect,
  useRef,
  useState,
} from 'react';

import Constants from 'expo-constants';
import { MainStackNavigationProps } from '../navigation/MainStackNavigator';
import { getString } from '../../../STRINGS';
import { isIPhoneXSize } from '../../utils/Styles';
import styled from 'styled-components/native';
import { useThemeContext } from '@dooboo-ui/native-theme';

const InnerContainer = styled.View`
  padding: 0 24px;
  flex: 1;
  width: 100%;
`;

const StyledKeyboardAvoidingView = styled.KeyboardAvoidingView`
  flex: 1;
  justify-content: center;
  align-self: stretch;
  flex-direction: column;
  align-items: center;
`;

export interface Props {
  navigation: MainStackNavigationProps<'ChangePw'>;
}

function ChangePw(props: Props): ReactElement {
  const { navigation } = props;
  const { theme } = useThemeContext();
  const [currentPw, setCurrentPw] = useState('');
  const [newPw, setNewPw] = useState('');
  const [confirmPw, setConfirmPw] = useState('');
  const close = (): void => {
    navigation.goBack();
  };

  const changePassword = async (): Promise<void> => {
    if (newPw === confirmPw) {
      // TODO change password api call
      Keyboard.dismiss();
      Alert.alert('', 'Password changed.', [
        {
          text: getString('OK'),
          onPress: (): void => {
            close();
          },
        },
      ]);
    } else {
      Alert.alert(getString('PASSWORD_MUST_MATCH'));
    }
  };
  const [keyboardOffset, setKeyboardOffset] = useState(0);
  const onKeyboardShow = (event: KeyboardEvent): void => setKeyboardOffset(event.endCoordinates.height);
  const onKeyboardHide = (): void => setKeyboardOffset(0);
  const keyboardDidShowListener = useRef<EmitterSubscription>();
  const keyboardDidHideListener = useRef<EmitterSubscription>();

  useEffect(() => {
    keyboardDidShowListener.current = Keyboard.addListener('keyboardWillShow', onKeyboardShow);
    keyboardDidHideListener.current = Keyboard.addListener('keyboardWillHide', onKeyboardHide);

    return (): void => {
      keyboardDidShowListener.current && keyboardDidShowListener.current.remove();
      keyboardDidHideListener.current && keyboardDidHideListener.current.remove();
    };
  }, []);
  return (
    <SafeAreaView
      style={{
        flex: 1,
        backgroundColor: theme.background,
        paddingBottom: keyboardOffset,
      }}>
      <StyledKeyboardAvoidingView
        keyboardVerticalOffset={isIPhoneXSize
          ? Constants.statusBarHeight + 52
          : Constants.statusBarHeight}
        behavior={'padding'}
      >
        <InnerContainer>
          <EditText
            key="input-pw"
            testID="input-pw"
            style={{ marginTop: 40 }}
            borderColor={theme.font}
            focusColor={theme.focused}
            placeholderTextColor={theme.placeholder}
            secureTextEntry
            onChangeText={(pw: string): void => setCurrentPw(pw)}
            label={getString('PASSWORD_CURRENT')}
            value={currentPw}
            placeholder="******"
          />
          <EditText
            key="new-pw-input"
            testID="new-pw-input"
            style={{ marginTop: 20 }}
            secureTextEntry
            borderColor={theme.font}
            focusColor={theme.focused}
            placeholderTextColor={theme.placeholder}
            onChangeText={(pw: string): void => setNewPw(pw)}
            label={getString('PASSWORD_NEW')}
            value={newPw}
            placeholder="******"
          />
          <EditText
            key="input-validation"
            testID="input-validation"
            style={{ marginTop: 20 }}
            secureTextEntry
            borderColor={theme.font}
            focusColor={theme.focused}
            placeholderTextColor={theme.placeholder}
            onChangeText={(pw: string): void => setConfirmPw(pw)}
            label={getString('PASSWORD_NEW_REPEAT')}
            value={confirmPw}
            placeholder="******"
          />
        </InnerContainer>
        <Button
          testID="close-current-pw-btn"
          onPress={changePassword}
          containerStyle={{
            width: '100%',
            paddingHorizontal: 20,
          }}
          style={{
            backgroundColor: theme.btnPrimary,
            borderWidth: 0,
            height: 48,
            width: '100%',
          }}
          textStyle={{
            color: theme.btnPrimaryFont,
            fontSize: 16,
          }}
          text={getString('UPDATE')}
        />
      </StyledKeyboardAvoidingView>
    </SafeAreaView>
  );
}

export default ChangePw;
