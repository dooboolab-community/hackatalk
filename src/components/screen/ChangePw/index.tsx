import { Alert, EmitterSubscription, Keyboard, KeyboardEvent, SafeAreaView } from 'react-native';
import {
  CloseButton,
  Header,
  InnerContainer,
  StyledTextInput,
  Title,
} from './styles';
import React, {
  ReactElement,
  useEffect,
  useRef,
  useState,
} from 'react';

import { Button } from '@dooboo-ui/native';
import Constants from 'expo-constants';
import { Ionicons } from '@expo/vector-icons';
import { MainStackNavigationProps } from '../../navigation/MainStackNavigator';
import { getString } from '../../../../STRINGS';
import { isIPhoneXSize } from '../../../utils/Styles';
import styled from 'styled-components/native';
import { useThemeContext } from '@dooboo-ui/native-theme';

export interface Props {
  navigation: MainStackNavigationProps<'ChangePw'>;
}

const StyledKeyboardAvoidingView = styled.KeyboardAvoidingView`
  flex: 1;
  justify-content: center;
  align-self: stretch;
  flex-direction: column;
  align-items: center;
`;

const ChangePwHeader = (props: Props): ReactElement => {
  const { navigation } = props;
  const { theme } = useThemeContext();
  return (
    <Header>
      <Title>{getString('PASSWORD_CHANGE')}</Title>
      <CloseButton testID="closeBtn" onPress={(): void => navigation.goBack()}>
        <Ionicons name="md-close" size={24} color={theme.fontColor} />
      </CloseButton>
    </Header>
  );
};

function ChangePw(props: Props): ReactElement {
  const { navigation } = props;
  const { theme } = useThemeContext();
  const [isValidCurrentPw, setCurrentPwValid] = useState(false);
  const [currentPw, setCurrentPw] = useState('');
  const [newPw, setNewPw] = useState('');
  const [validationWord, setValidationWord] = useState('');
  const close = (): void => {
    navigation.goBack();
  };
  const validateCurrent = (): void => {
    try {
      if (currentPw === 'right') {
        setCurrentPwValid(true);
      } else {
        throw Error(getString('PASSWORD_MUST_MATCH'));
      }
    } catch (err) {
      Alert.alert('', err.message);
    }
  };
  const changePassword = async (): Promise<void> => {
    if (newPw === validationWord) {
      // TODO change password api call
      Keyboard.dismiss();
      Alert.alert('', 'Password changed.', [
        {
          text: getString('OK'),
          onPress: (): void => {
            close();
          },
        },
      ]);
    } else {
      Alert.alert(getString('PASSWORD_MUST_MATCH'));
    }
  };
  const [keyboardOffset, setKeyboardOffset] = useState(0);
  const onKeyboardShow = (event: KeyboardEvent): void => setKeyboardOffset(event.endCoordinates.height);
  const onKeyboardHide = (): void => setKeyboardOffset(0);
  const keyboardDidShowListener = useRef<EmitterSubscription>();
  const keyboardDidHideListener = useRef<EmitterSubscription>();

  useEffect(() => {
    keyboardDidShowListener.current = Keyboard.addListener('keyboardWillShow', onKeyboardShow);
    keyboardDidHideListener.current = Keyboard.addListener('keyboardWillHide', onKeyboardHide);

    return (): void => {
      keyboardDidShowListener.current && keyboardDidShowListener.current.remove();
      keyboardDidHideListener.current && keyboardDidHideListener.current.remove();
    };
  }, []);
  return (
    <SafeAreaView
      style={{
        flex: 1,
        backgroundColor: theme.background,
        paddingBottom: keyboardOffset,
      }}>
      <StyledKeyboardAvoidingView
        keyboardVerticalOffset={isIPhoneXSize
          ? Constants.statusBarHeight + 52
          : Constants.statusBarHeight}
        behavior={'padding'}
      >
        {isValidCurrentPw ? (
          <InnerContainer>
            <StyledTextInput
              key="newPwTextInput"
              testID="newPwTextInput"
              isPassword
              onTextChanged={(pw): void => setNewPw(pw)}
              txtLabel={getString('PASSWORD_NEW')}
            />
            <StyledTextInput
              key="validationWordTextInput"
              testID="validationWordTextInput"
              isPassword
              onTextChanged={(pw): void => setValidationWord(pw)}
              txtLabel={getString('PASSWORD_NEW_REPEAT')}
            />
          </InnerContainer>
        ) : (
          <InnerContainer>
            <StyledTextInput
              key="currentPwTextInput"
              testID="currentPwTextInput"
              isPassword
              onTextChanged={(pw): void => setCurrentPw(pw)}
              txtLabel={getString('PASSWORD_CURRENT')}
            />
          </InnerContainer>
        )}
        <Button
          testID="checkCurrentPwBtn"
          onPress={isValidCurrentPw ? changePassword : validateCurrent}
          containerStyle={{
            width: '100%',
          }}
          style={{
            backgroundColor: theme.btnPrimary,
            borderWidth: 0,
            height: 56,
            width: '100%',
          }}
          textStyle={{
            color: theme.btnPrimaryFont,
            fontSize: 16,
          }}
          text={isValidCurrentPw ? getString('CONFIRM') : getString('NEXT')}
        />
      </StyledKeyboardAvoidingView>
    </SafeAreaView>
  );
}

// eslint-disable-next-line
export const ChangePwHeaderOptions = (): { header: any } => ({ header: ChangePwHeader });
export default ChangePw;
