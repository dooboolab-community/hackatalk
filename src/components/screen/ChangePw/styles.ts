import Constants from 'expo-constants';
import TextInput from '../../shared/TextInput';
import styled from 'styled-components/native';

export const Header = styled.View`
  height: ${52 + Constants.statusBarHeight}px;
  padding-top: ${Constants.statusBarHeight}px;
  background-color: ${({ theme }): string => theme.background};
  justify-content: center;
  align-items: center;
`;

export const Title = styled.Text`
  color: ${({ theme }): string => theme.fontColor};
  font-size: 20px;
`;

export const CloseButton = styled.TouchableOpacity`
  height: 52px;
  padding: 8px;
  position: absolute;
  right: 10px;
  top: ${Constants.statusBarHeight}px;
  justify-content: center;
`;

export const InnerContainer = styled.View`
  padding: 0 24px;
  flex: 1;
  width: 100%;
`;

export const StyledTextInput = styled(TextInput).attrs(({ theme }) => ({
  placeholderTextColor: theme.placeholder,
}))``;
