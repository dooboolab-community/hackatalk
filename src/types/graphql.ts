export type Maybe<T> = T | null;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  Date: any;
  DateTime: any;
  Upload: any;
};

export type AuthPayload = {
  __typename?: 'AuthPayload';
  token: Scalars['String'];
  user: User;
};

export enum AuthType {
  Email = 'email',
  Facebook = 'facebook',
  Google = 'google',
  Apple = 'apple'
}

export type Channel = {
  __typename?: 'Channel';
  id: Scalars['ID'];
  type?: Maybe<ChannelType>;
  name?: Maybe<Scalars['String']>;
  messages?: Maybe<Array<Maybe<Message>>>;
  memberships?: Maybe<Array<Maybe<Membership>>>;
  myMembership?: Maybe<Membership>;
  createdAt: Scalars['DateTime'];
  updatedAt: Scalars['DateTime'];
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export type ChannelInput = {
  type?: Maybe<ChannelType>;
  name?: Maybe<Scalars['String']>;
  friendIds: Array<Scalars['String']>;
};

export enum ChannelType {
  Private = 'PRIVATE',
  Public = 'PUBLIC'
}



export type File = {
  __typename?: 'File';
  filename: Scalars['String'];
  mimetype: Scalars['String'];
  encoding: Scalars['String'];
};

export type Friend = {
  __typename?: 'Friend';
  id: Scalars['ID'];
  user?: Maybe<User>;
  friend?: Maybe<User>;
  createdAt?: Maybe<Scalars['DateTime']>;
  updatedAt?: Maybe<Scalars['DateTime']>;
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export type FriendPayload = {
  __typename?: 'FriendPayload';
  user: User;
  added?: Maybe<Scalars['Boolean']>;
  deleted?: Maybe<Scalars['Int']>;
};

export type FriendSub = {
  __typename?: 'FriendSub';
  user?: Maybe<User>;
  action?: Maybe<FriendSubAction>;
};

export enum FriendSubAction {
  Added = 'ADDED',
  Updated = 'UPDATED',
  Deleted = 'DELETED'
}

export type Gallery = {
  __typename?: 'Gallery';
  id: Scalars['ID'];
  photoURL: Scalars['String'];
  createdAt: Scalars['DateTime'];
  updatedAt: Scalars['DateTime'];
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export enum Gender {
  Male = 'MALE',
  Female = 'FEMALE'
}

export type Membership = {
  __typename?: 'Membership';
  id: Scalars['ID'];
  channel?: Maybe<Channel>;
  user?: Maybe<User>;
  type?: Maybe<MemberType>;
  userAlert?: Maybe<Scalars['Boolean']>;
  userMode?: Maybe<UserModeType>;
  createdAt?: Maybe<Scalars['DateTime']>;
  updatedAt?: Maybe<Scalars['DateTime']>;
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export enum MemberType {
  Owner = 'OWNER',
  Member = 'MEMBER'
}

export type Message = {
  __typename?: 'Message';
  id: Scalars['String'];
  channel?: Maybe<Channel>;
  sender?: Maybe<User>;
  type?: Maybe<Scalars['String']>;
  text?: Maybe<Scalars['String']>;
  picture?: Maybe<Array<Maybe<Photo>>>;
  filePath?: Maybe<Scalars['String']>;
  replies?: Maybe<Array<Maybe<Reply>>>;
  reactions?: Maybe<Array<Maybe<Reaction>>>;
  createdAt?: Maybe<Scalars['DateTime']>;
  updatedAt?: Maybe<Scalars['DateTime']>;
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export type MessagePayload = {
  __typename?: 'MessagePayload';
  channelId: Scalars['String'];
  message?: Maybe<Message>;
};

export type Mutation = {
  __typename?: 'Mutation';
  signInEmail: AuthPayload;
  signInWithSocialAccount: AuthPayload;
  signInWithFacebook: AuthPayload;
  signUp: AuthPayload;
  findPassword?: Maybe<Scalars['Boolean']>;
  sendVerification?: Maybe<Scalars['Boolean']>;
  addNotificationToken?: Maybe<Notification>;
  removeNotificationToken?: Maybe<Scalars['Int']>;
  updateProfile?: Maybe<User>;
  addFriend?: Maybe<FriendPayload>;
  deleteFriend?: Maybe<FriendPayload>;
  /** `friendIds` in Channel should exclude userid. */
  createChannel?: Maybe<Channel>;
  updateChannel?: Maybe<Scalars['Int']>;
  deleteChannel?: Maybe<Scalars['Int']>;
  /**
   * Create message and return channelId when meessage has successfully sent.
   * Do not pass current userId inside `users`.
   */
  createMessage?: Maybe<MessagePayload>;
  setOnlineStatus?: Maybe<Scalars['Int']>;
  changeEmailPassword?: Maybe<Scalars['Boolean']>;
  createGallery?: Maybe<Gallery>;
  updateGallery?: Maybe<Scalars['Int']>;
  deleteGallery?: Maybe<Scalars['Int']>;
  singleUpload: Scalars['String'];
  createReaction?: Maybe<Reaction>;
  deleteReaction?: Maybe<Scalars['Int']>;
};


export type MutationSignInEmailArgs = {
  email: Scalars['String'];
  password: Scalars['String'];
};


export type MutationSignInWithSocialAccountArgs = {
  socialUser: SocialUserInput;
};


export type MutationSignInWithFacebookArgs = {
  accessToken: Scalars['String'];
};


export type MutationSignUpArgs = {
  user: UserInput;
};


export type MutationFindPasswordArgs = {
  email: Scalars['String'];
};


export type MutationSendVerificationArgs = {
  email: Scalars['String'];
};


export type MutationAddNotificationTokenArgs = {
  notification: NotificationCreateInput;
};


export type MutationRemoveNotificationTokenArgs = {
  token: Scalars['String'];
};


export type MutationUpdateProfileArgs = {
  user: UserProfileInput;
};


export type MutationAddFriendArgs = {
  friendId: Scalars['ID'];
};


export type MutationDeleteFriendArgs = {
  friendId: Scalars['ID'];
};


export type MutationCreateChannelArgs = {
  channel?: Maybe<ChannelInput>;
};


export type MutationUpdateChannelArgs = {
  channel?: Maybe<ChannelInput>;
};


export type MutationDeleteChannelArgs = {
  channelId: Scalars['ID'];
};


export type MutationCreateMessageArgs = {
  message: Scalars['String'];
  channelId: Scalars['String'];
};


export type MutationSetOnlineStatusArgs = {
  isOnline?: Maybe<Scalars['Boolean']>;
};


export type MutationChangeEmailPasswordArgs = {
  password: Scalars['String'];
  newPassword: Scalars['String'];
};


export type MutationCreateGalleryArgs = {
  photoURL: Scalars['String'];
};


export type MutationUpdateGalleryArgs = {
  galleryId: Scalars['ID'];
  photoURL: Scalars['String'];
};


export type MutationDeleteGalleryArgs = {
  galleryId: Scalars['ID'];
};


export type MutationSingleUploadArgs = {
  file: Scalars['Upload'];
  dir?: Maybe<Scalars['String']>;
};


export type MutationCreateReactionArgs = {
  messageId: Scalars['ID'];
  type: Scalars['String'];
};


export type MutationDeleteReactionArgs = {
  reactionId: Scalars['ID'];
};

export type Notification = {
  __typename?: 'Notification';
  id: Scalars['ID'];
  token?: Maybe<Scalars['String']>;
  device?: Maybe<Scalars['String']>;
  os?: Maybe<Scalars['String']>;
  createdAt?: Maybe<Scalars['DateTime']>;
  updatedAt?: Maybe<Scalars['DateTime']>;
};

export type NotificationCreateInput = {
  token: Scalars['String'];
  device?: Maybe<Scalars['String']>;
  os?: Maybe<Scalars['String']>;
};

/** Information about pagination in a connection. */
export type PageInfo = {
  __typename?: 'PageInfo';
  startCursor?: Maybe<Scalars['String']>;
  endCursor?: Maybe<Scalars['String']>;
  hasNextPage?: Maybe<Scalars['Boolean']>;
  hasPreviousPage?: Maybe<Scalars['Boolean']>;
};

export type Photo = {
  __typename?: 'Photo';
  id: Scalars['String'];
  thumbURL?: Maybe<Scalars['String']>;
  photoURL?: Maybe<Scalars['String']>;
  createdAt?: Maybe<Scalars['DateTime']>;
  updatedAt?: Maybe<Scalars['DateTime']>;
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export type Query = {
  __typename?: 'Query';
  /**
   * If filter is true, it will filter user with email, nickname or name.
   * You can add pagination with first and after args.
   */
  users?: Maybe<UsersConnection>;
  user?: Maybe<User>;
  me?: Maybe<User>;
  messages: Array<Message>;
  channels: Array<Channel>;
  friends: Array<User>;
  galleries: Array<Gallery>;
};


export type QueryUsersArgs = {
  user?: Maybe<UserQueryInput>;
  includeUser?: Maybe<Scalars['Boolean']>;
  filter?: Maybe<Scalars['Boolean']>;
  first?: Maybe<Scalars['Int']>;
  last?: Maybe<Scalars['Int']>;
  before?: Maybe<Scalars['String']>;
  after?: Maybe<Scalars['String']>;
};


export type QueryUserArgs = {
  id: Scalars['ID'];
};


export type QueryGalleriesArgs = {
  userId: Scalars['String'];
};

export type Reaction = {
  __typename?: 'Reaction';
  id: Scalars['ID'];
  type?: Maybe<Scalars['String']>;
  createdAt?: Maybe<Scalars['DateTime']>;
  updatedAt?: Maybe<Scalars['DateTime']>;
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export type Reply = {
  __typename?: 'Reply';
  id: Scalars['String'];
  sender?: Maybe<User>;
  type?: Maybe<Scalars['String']>;
  text?: Maybe<Scalars['String']>;
  filePath?: Maybe<Scalars['String']>;
  replies?: Maybe<Array<Maybe<Reply>>>;
  createdAt?: Maybe<Scalars['DateTime']>;
  updatedAt?: Maybe<Scalars['DateTime']>;
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export type SocialUserInput = {
  socialId: Scalars['String'];
  authType: AuthType;
  email?: Maybe<Scalars['String']>;
  photoURL?: Maybe<Scalars['String']>;
  thumbURL?: Maybe<Scalars['String']>;
  name?: Maybe<Scalars['String']>;
  nickname?: Maybe<Scalars['String']>;
  birthday?: Maybe<Scalars['Date']>;
  gender?: Maybe<Gender>;
  phone?: Maybe<Scalars['String']>;
};

export type Subscription = {
  __typename?: 'Subscription';
  userSignedIn?: Maybe<User>;
  userUpdated?: Maybe<User>;
  friendChanged?: Maybe<FriendSub>;
};


export type SubscriptionUserSignedInArgs = {
  userId: Scalars['ID'];
};


export type SubscriptionUserUpdatedArgs = {
  userId: Scalars['ID'];
};


export type SubscriptionFriendChangedArgs = {
  userId: Scalars['ID'];
};


export type User = {
  __typename?: 'User';
  id: Scalars['ID'];
  email?: Maybe<Scalars['String']>;
  name?: Maybe<Scalars['String']>;
  nickname?: Maybe<Scalars['String']>;
  thumbURL?: Maybe<Scalars['String']>;
  photoURL?: Maybe<Scalars['String']>;
  birthday?: Maybe<Scalars['Date']>;
  gender?: Maybe<Gender>;
  socialId?: Maybe<Scalars['String']>;
  authType?: Maybe<AuthType>;
  phone?: Maybe<Scalars['String']>;
  verified?: Maybe<Scalars['Boolean']>;
  statusMessage?: Maybe<Scalars['String']>;
  isOnline?: Maybe<Scalars['Boolean']>;
  lastSignedIn?: Maybe<Scalars['DateTime']>;
  notifications?: Maybe<Array<Maybe<Notification>>>;
  createdAt?: Maybe<Scalars['DateTime']>;
  updatedAt?: Maybe<Scalars['DateTime']>;
  deletedAt?: Maybe<Scalars['DateTime']>;
};

export type UserEdge = {
  __typename?: 'UserEdge';
  node?: Maybe<User>;
  cursor?: Maybe<Scalars['String']>;
};

export type UserInput = {
  email: Scalars['String'];
  password: Scalars['String'];
  name?: Maybe<Scalars['String']>;
  nickname?: Maybe<Scalars['String']>;
  birthday?: Maybe<Scalars['Date']>;
  gender?: Maybe<Gender>;
  phone?: Maybe<Scalars['String']>;
  statusMessage?: Maybe<Scalars['String']>;
};

export enum UserModeType {
  Default = 'DEFAULT',
  Hidden = 'HIDDEN',
  Block = 'BLOCK'
}

export type UserProfileInput = {
  name?: Maybe<Scalars['String']>;
  nickname?: Maybe<Scalars['String']>;
  birthday?: Maybe<Scalars['Date']>;
  gender?: Maybe<Gender>;
  phone?: Maybe<Scalars['String']>;
  thumbURL?: Maybe<Scalars['String']>;
  photoURL?: Maybe<Scalars['String']>;
  statusMessage?: Maybe<Scalars['String']>;
};

export type UserQueryInput = {
  email?: Maybe<Scalars['String']>;
  name?: Maybe<Scalars['String']>;
  nickname?: Maybe<Scalars['String']>;
  birthday?: Maybe<Scalars['Date']>;
  gender?: Maybe<Gender>;
  phone?: Maybe<Scalars['String']>;
};

/**
 * Simple wrapper around our list of launches that contains a cursor to the
 * last item in the list. Pass this cursor to the launches query to fetch results
 * after these.
 */
export type UsersConnection = {
  __typename?: 'UsersConnection';
  totalCount?: Maybe<Scalars['Int']>;
  edges?: Maybe<Array<Maybe<UserEdge>>>;
  pageInfo?: Maybe<PageInfo>;
};
