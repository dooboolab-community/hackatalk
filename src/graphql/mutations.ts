import { User } from '../types';
import gql from 'graphql-tag';

export interface MutationSignUpInput {
  user: {
    email: string;
    password: string;
    name: string;
    statusMessage: string;
  };
}

export interface AuthPayload {
  token: string;
  user: User;
}

export const MUTATION_SIGN_UP = gql`
  mutation signUp($user: UserInput!) {
    signUp(user: $user) {
      token
      user {
        id
        email
        verified
      }
    }
  }
`;

export interface MutationSendVerificationInput {
  email: string;
}

export const MUTATION_SEND_VERIFICATION = gql`
  mutation sendVerification($email: String!) {
    sendVerification(email: $email)
  }
`;

export interface SignInEmailInput {
  email: string; password: string;
}

export const MUTATION_SIGN_IN = gql`
  mutation signInEmail($email: String! $password: String!) {
    signInEmail(email: $email password: $password) {
      token
      user {
        id
        email
        nickname
        verified
        statusMessage
        authType
      }
    }
  }
`;

export interface MutationChangePasswordInput {
  currentPassword: string;
  newPassword: string;
}

export const MUTATION_CHANGE_PASSWORD = gql`
  mutation changeEmailPassword($currentPassword: String! $newPassword: String!) {
    changeEmailPassword(password: $currentPassword newPassword: $newPassword) {
      changeEmailPassword
    }
  }
`;

export interface MutationFindPasswordInput {
  email: string;
}

export const MUTATION_FIND_PASSWORD = gql`
  mutation findPassword($email: String!) {
    findPassword(email: $email) {
      findPassword
    }
  }
`;

export interface AddOrDeleteFriendInput {
  friendId: string;
}

export interface FriendPayload {
  user: User;
  added?: boolean;
  deleted?: boolean;
}

export const MUTATION_ADD_FRIEND = gql`
  mutation addFriend($friendId: ID!){
    addFriend(friendId: $friendId) {
      user {
        id
        email
        name
      }
      added
    }
  }
`;

export const MUTATION_DELETE_FRIEND = gql`
  mutation deleteFriend($friendId: ID!) {
    deleteFriend(friendId: $friendId) {
      user {
        id
      }
      deleted
    }
  }
`;
