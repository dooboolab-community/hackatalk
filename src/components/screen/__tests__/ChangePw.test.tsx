import * as React from 'react';

import {
  RenderResult,
  act,
  cleanup,
  fireEvent,
  render,
  wait,
} from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import ChangePw from '../ChangePw';
import { MUTATION_CHANGE_PASSWORD } from '../../../graphql/mutations';
import { getString } from '../../../../STRINGS';

const mockChangePwMutation = [
  {
    request: {
      query: MUTATION_CHANGE_PASSWORD,
      variables: {
        currentPassword: 'currentPassword',
        newPassword: 'newPassword',
      },
    },
    error: new Error('Error occurred'),
  },
  {
    request: {
      query: MUTATION_CHANGE_PASSWORD,
      variables: {
        currentPassword: 'currentPassword',
        newPassword: 'newPassword',
      },
    },
    result: {
      data: {
        changeEmailPassword: true,
      },
    },
  },
];

const mockAlert = {
  alert: jest.fn(),
};

const mockKeyboard = {
  addListener: jest.fn(),
  dismiss: jest.fn(),
};

jest.mock('react-native', () => {
  const ReactNative = jest.requireActual('react-native');
  Object.defineProperty(ReactNative, 'Alert', {
    get: () => mockAlert,
  });
  Object.defineProperty(ReactNative, 'Keyboard', {
    get: () => mockKeyboard,
  });
  return ReactNative;
});

describe('[ChangePw] screen', () => {
  let testingLib: RenderResult;

  beforeEach(() => {
    const props = createTestProps({});
    const component = createTestElement(
      <ChangePw {...props} />,
    );
    testingLib = render(component);
    mockAlert.alert.mockClear();
  });

  it('renders without crashing', () => {
    expect(testingLib.asJSON()).toMatchSnapshot();
    expect(testingLib.asJSON()).toBeTruthy();
  });

  describe('interactions', () => {
    it('should simulate textChanged', async () => {
      const pwInput = testingLib.getByTestId('input-pw');
      const verifyBtn = testingLib.getByTestId('input-pw');
      act(() => {
        fireEvent.changeText(pwInput, 'left');
      });
      act(() => {
        fireEvent.press(verifyBtn);
      });
      act(() => {
        fireEvent.changeText(pwInput, 'right');
      });
      act(() => {
        fireEvent.press(verifyBtn);
      });

      const keyboardEvents = mockKeyboard.addListener.mock.calls;
      keyboardEvents.forEach(
        async ([eventName, ftn]: [string, (arg?: any) => void]) => {
          switch (eventName) {
            case 'keyboardWillShow':
              act(() => ftn({ endCoordinates: { height: 301 } }));
              break;
            case 'keyboardWillHide':
              act(() => ftn());
              break;
            default:
          }
        },
      );
    });
  });

  afterAll(() => {
    cleanup();
  });
});
