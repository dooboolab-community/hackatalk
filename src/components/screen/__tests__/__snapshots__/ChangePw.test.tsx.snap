// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ChangePw] screen renders without crashing 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "alignSelf": "stretch",
              "flexDirection": "column",
              "justifyContent": "flex-start",
            },
          ]
        }
      >
        <Text
          focused={false}
          style={
            Array [
              Object {
                "color": "#a3a3a3",
                "fontSize": 12,
                "marginBottom": 8,
              },
            ]
          }
        >
          Enter Current Password
        </Text>
        <View
          isFocused={false}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "rgb(221,226,236)",
                "borderRadius": 3,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 8,
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            focused={false}
            placeholderTextColor="#B9B9C4"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "borderColor": "black",
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "color": "black",
                  "flexBasis": 0,
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 16,
                  "justifyContent": "flex-start",
                  "paddingBottom": 22,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 22,
                },
              ]
            }
            testID="currentPwTextInput"
            textContentType="none"
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    </View>
    <TouchableOpacity
      activeOpacity={0.5}
      disabled={false}
      height={60}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(58,139,255)",
            "borderColor": "white",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 60,
            "justifyContent": "center",
            "width": "auto",
          },
        ]
      }
      testID="checkCurrentPwBtn"
      white={false}
      width="auto"
    >
      <Text
        disabled={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 14,
              "fontWeight": "bold",
            },
          ]
        }
        white={false}
      >
        Next
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`[ChangePwHeader] component renders without crashing 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "height": 96,
          "justifyContent": "center",
          "paddingTop": 44,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 20,
          },
        ]
      }
    >
      Change Password
    </Text>
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Array [
          Object {
            "height": 52,
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
            "position": "absolute",
            "right": 10,
            "top": 44,
          },
        ]
      }
      testID="closeBtn"
    >
      <Text />
    </TouchableOpacity>
  </View>
</View>
`;
